rotkehlchen/chain/evm/decoding/hop/balances.py

```diff
@@ -0,0 +1,165 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import token_normalized_value
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.decoding.hop.constants import CPT_HOP
+from rotkehlchen.chain.evm.tokens import get_chunk_size_call_order
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.evm_event import EvmProduct
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class HopCommonBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: 'DBHandler',
+            evm_inquirer: 'EvmNodeInquirer',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            counterparty=CPT_HOP,
+            deposit_event_types={(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+
+    def query_balances(self) -> 'BalancesSheetType':
+        """Queries and returns the balance sheet for staking events.
+
+        This method:
+        1. Retrieves deposit events for staking.
+        2. Groups addresses by staking contract.
+        3. For each contract, queries user balances and rewards.
+        4. Converts balances and rewards to USD and updates the balance sheet."""
```

Comment:
nitpick: Can you be putting the end ofthe docstring at a newline if it's a big multiline description? Makes it look more neat and separates from the body

```suggestion
        4. Converts balances and rewards to USD and updates the balance sheet.
        """
```